src/services/SessionRequestProcessor.ts

```diff
@@ -0,0 +1,213 @@
+  	const requestBodyClientId = deserialisedRequestBody.client_id;
+  	const clientIpAddress = event.requestContext.identity?.sourceIp;
+
+  	let configClient: ClientConfig | undefined;
+  	try {
+  		const config = JSON.parse(this.clientConfig) as ClientConfig[];
+  		configClient = config.find(c => c.clientId === requestBodyClientId);
+  	} catch (error: any) {
+  		this.logger.error("Invalid or missing client configuration table", {
+  			error,
+  			messageCode: MessageCodes.MISSING_CONFIGURATION,
+  		});
+  		return new Response(HttpCodesEnum.SERVER_ERROR, "Server Error");
+  	}
+
+  	if (!configClient) {
+  		this.logger.error("Unrecognised client in request", {
+  			messageCode: MessageCodes.UNRECOGNISED_CLIENT,
+  		});
+  		return new Response(HttpCodesEnum.BAD_REQUEST, "Bad Request");
+  	}
+
+  	let urlEncodedJwt: string;
+  	try {
+  		urlEncodedJwt = await this.kmsDecryptor.decrypt(deserialisedRequestBody.request);
+  	} catch (error: any) {
+  		this.logger.error("Failed to decrypt supplied JWE request", {
+  			error,
+  			messageCode: MessageCodes.FAILED_DECRYPTING_JWE,
+  		});
+  		return UnauthorizedResponse;
+  	}
+
+  	let parsedJwt: Jwt;
+  	try {
+  		parsedJwt = this.kmsDecryptor.decode(urlEncodedJwt);
+  	} catch (error: any) {
+  		this.logger.error("Failed to decode supplied JWT", {
+  			error,
+  			messageCode: MessageCodes.FAILED_DECODING_JWT,
+  		});
+  		return UnauthorizedResponse;
+  	}
+
+  	try {
+  		if (configClient.jwksEndpoint) {
+  			const payload = await this.kmsDecryptor.verifyWithJwks(urlEncodedJwt, configClient.jwksEndpoint);
+
+  			if (!payload) {
+  				this.logger.error("Failed to verify JWT", {
+  					messageCode: MessageCodes.FAILED_VERIFYING_JWT,
+  				});
+  				return UnauthorizedResponse;
+  			}
+  		} else {
+  			this.logger.error("Incomplete Client Configuration", {
+  				messageCode: MessageCodes.MISSING_CONFIGURATION,
+  			});
+  			return new Response(HttpCodesEnum.SERVER_ERROR, "Server Error");
+  		}
+  	} catch (error: any) {
+  		this.logger.error("Invalid request: Could not verify jwt", {
+  			error,
+  			messageCode: MessageCodes.FAILED_VERIFYING_JWT,
+  		});
+  		return UnauthorizedResponse;
+  	}
+
+  	const jwtPayload: JwtPayload = parsedJwt.payload;
+  	const JwtErrors = isJwtValid(jwtPayload, requestBodyClientId, configClient.redirectUri);
+  	if (JwtErrors.length > 0) {
+  		this.logger.error(JwtErrors, {
+  			messageCode: MessageCodes.FAILED_VALIDATING_JWT,
+  		});
+  		return UnauthorizedResponse;
+  	}
+
+  	const personDetailsError = isPersonNameValid(jwtPayload.shared_claims.name);
+  	if (!personDetailsError) {
+  		this.logger.error({
+  			message: "Missing GivenName or FamilyName from shared claims data",
+  			messageCode: MessageCodes.INVALID_PERSONAL_DETAILS,
+  		});
+  		return UnauthorizedResponse;
+  	}
+
+  	const sessionId: string = await this.BavService.generateSessionId();
+  	this.logger.appendKeys({
```

Comment:
Could we append govuk_signin_journey_id before the nameCheck? So we have something to track by should we get JWTs with missing names

Reply:
Yeah good shout, done